### PR TITLE
ISPN-15164 RemoteStore can ignore defined configuration properties

### PR DESCRIPTION
**Pull request by srose**
Fixes https://issues.redhat.com/browse/ISPN-15164.

Originally: Alternative solution to the changes from https://issues.redhat.com/browse/ISPN-13310.

### CODE REVIEW

File: persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java [hunk 1]
```diff
@@ -514,16 +514,9 @@ private ConfigurationBuilder buildRemoteConfiguration(RemoteStoreConfiguration c
          ;
       }
 
-      Properties propertiesToUse;
-      Properties actualProperties = configuration.properties();
-      if (!actualProperties.contains("blocking")) {
```
Comment (srose) [inline persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:519]: checks for a value not a key

File: persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java [hunk 2]
```diff
@@ -514,16 +514,9 @@ private ConfigurationBuilder buildRemoteConfiguration(RemoteStoreConfiguration c
...
-         // Need to make a copy to not change the actual configuration properties
-         propertiesToUse = new Properties(actualProperties);
```
Comment (srose) [inline persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:521]: initializer with default values, requires consumers to fetch values not check if they are present

File: persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java [hunk 3]
```diff
@@ -514,16 +514,9 @@ private ConfigurationBuilder buildRemoteConfiguration(RemoteStoreConfiguration c
...
-         // Make sure threads are marked as non blocking if user didn't specify
-         propertiesToUse.put("blocking", "false");
```
Comment (srose) [inline persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java:523]: inserts a key

**Comment by thomasdarimont** [comment]
LGTM

**Comment by tristantarrant** [comment]
@wburns can you have a look?

**Comment by wburns** [comment]
This should be logged as a separate bug imo as the original ISPN-13310 breaks passing any properties to the underlying remote cache manager as https://github.com/infinispan/infinispan/blob/main/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java#L436 only writes the properties explicitly written, not the default ones.

The question is how do we fix. The change here works great, but I wonder if TypedProperties may be missing other locations that utilize default properties and we should make sure those are copied as well.

**Comment by srose** [comment]
@wburns thanks, i create a new issue and link the issue there.
 I need more time to think / more details on the remark regarding typed properties and defaults.

**Comment by wburns** [comment]
Updated commit message and added an explicit test. Waiting on CI to integrate.

Unfortunately, java Properties do not have any way to easily access the default Properties underneath which seems bad to me, but we can just stick with the changes here.

**Comment by wburns** [comment]
Integrated into main, thanks @srose !

**Pull request merged**